embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/EmbraceUrl.kt

```diff
@@ -1,40 +1,38 @@
 package io.embrace.android.embracesdk.comms.api
 
 import java.io.IOException
+import java.net.HttpURLConnection
 import java.net.MalformedURLException
+import java.net.URL
 
-internal abstract class EmbraceUrl {
+internal class EmbraceUrl(val url: URL) {
```

Comment:
This class contains a lot of vestigial complexity like the `UrlFactory` that isn't needed. Remove them so we don't have untested dead code.